### PR TITLE
update version in dubbo-samples-validation

### PR DESCRIPTION
**Pull request by mming404**
update the jdk version to 17 ,dubbo version to 3.2.*,use springboot replace spring.

### CODE REVIEW

File: 2-advanced/dubbo-samples-validation/pom.xml [hunk 1]
```diff
@@ -90,65 +83,100 @@
...
+                </exclusion>
+                <exclusion>
+                    <artifactId>log4j</artifactId>
+                    <groupId>log4j</groupId>
+                </exclusion>
+                <exclusion>
+                    <artifactId>slf4j-log4j12</artifactId>
+                    <groupId>org.slf4j</groupId>
+                </exclusion>
+            </exclusions>
```
Comment (chickenlj) [inline 2-advanced/dubbo-samples-validation/pom.xml:106]: One small issue, please use the dependency listed here instead https://github.com/apache/dubbo-samples/blob/master/2-advanced/dubbo-samples-mock/dubbo-samples-mock-provider/pom.xml#L47

**Review by chickenlj — CHANGES_REQUESTED** [review]

**Review by chickenlj — APPROVED** [review]
LGTM.

**Comment by mming404** [comment]
can you merge it?please @chickenlj

File: 2-advanced/dubbo-samples-validation/src/test/java/org/apache/dubbo/samples/validation/ValidationServiceIT.java [hunk 2]
```diff
@@ -17,37 +17,41 @@
...
 
-@RunWith(SpringJUnit4ClassRunner.class)
-@ContextConfiguration(locations = {"classpath:/spring/validation-consumer.xml"})
+@SpringBootTest(classes = {ValidationConsumer.class})
+@RunWith(SpringRunner.class)
+@ExtendWith(SpringExtension.class)
 public class ValidationServiceIT {
-    @Autowired
+
+    @DubboReference(url = "dubbo://localhost:20880")
```
Comment (chickenlj) [inline 2-advanced/dubbo-samples-validation/src/test/java/org/apache/dubbo/samples/validation/ValidationServiceIT.java:39]: `@DubboReference` would be ok, no `url` parameter needed.

File: 2-advanced/dubbo-samples-validation/src/main/resources/application.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        name: validation-provider
+dubbo:
+    application:
+        qos-enable: false
+    protocol:
+        port: -1
+    consumer:
+        validation: true
+    provider:
+        validation: true
```
Comment (chickenlj) [inline 2-advanced/dubbo-samples-validation/src/main/resources/application.yml:29]: Please set `registry` here and change case-configuration.yml accordingly by referencing the following demo:
https://github.com/apache/dubbo-samples/tree/master/2-advanced/dubbo-samples-mock

**Review by chickenlj — CHANGES_REQUESTED** [review]

**Review by chickenlj — APPROVED** [review]
Well done.

**Pull request merged**